bot/exts/affinity_working_groups.py

```diff
@@ -1,162 +1,105 @@
-        await _add_remove_from_group(ctx, affinity_group, category, False)
-
-
-async def _has_role_check(ctx: commands.Context, guild, *role_name: int) -> bool:
-    """Checks if the user has a specific role within the server"""
-    member = guild.get_member(int(ctx.author.id))
-
-    for role in member.roles:
-        if role.name in role_name:
-            return True
-    return False
-
-
-async def _add_remove_from_group(ctx: commands.Context, channel_name: str, category,  add: bool = True) -> bool:
-    """Helper function to add or remove a user from a specific channel.
-    If add is false, it will remove the permissions."""
-
-    guild = discord.utils.get(bot.guilds, name=ACTIVE_GUILD)
-
-    active_group = ''
-    if category.name == "affinity groups":
-        active_group = AFFINITY_GROUPS
-    elif category.name == "Working Groups":
-        active_group = WORKING_GROUPS
-
-    msg_content = f":x: Sorry, there was an error. Please try again or contact an admin."
-
-    for key, value in active_group.items():
-        if channel_name.lower() in value:
-            channel = discord.utils.get(guild.channels, category_id=category.id, name=key)
-            await channel.set_permissions(ctx.author, read_messages=add, send_messages=add, add_reactions=add,
-                                          read_message_history=add, external_emojis=add, attach_files=add,
-                                          embed_links=add)
-            if add is True:
-                result = "added"
-            else:
-                result = "removed"
-            msg_content = f":white_check_mark: You have been successfully {result} from the {key} affinity group channel.\
-                \nIf there was an error, please contact an admin."
-    await ctx.author.send(msg_content)
-    if ctx.channel.type is discord.ChannelType.text:
-        await ctx.message.delete()
-
-
-@bot.command(hidden=True)
-async def on_ready(self, *args, **kwargs):
-    print(f'{bot.user} has connected to Discord')
-
-
-bot.add_cog(AffinityGroups(bot))
-bot.add_cog(WorkingGroups(bot))
-
-bot.run(TOKEN)
+import discord
+from discord.ext import commands
+
+from bot.bot import bot
+from bot.constants import ACTIVE_GUILD, AFFINITY_GROUPS, WORKING_GROUPS
+from bot.utils.role_checks import _has_role_check
+
+
+class WorkingGroups(commands.Cog):
+    """A cog that manages users adding themslves and leaving Working Groups"""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @commands.command()
+    async def join_wg(self, ctx: commands.Context, *, working_group: str) -> None:
+        """Direct Message the bot with the affinity group they want to join. The bot will add you to the correct group.
+
+        The current working groups can be found on the oSTEM.org website.
+        """
+        guild = discord.utils.get(bot.guilds, name=ACTIVE_GUILD)
+        if await _has_role_check(ctx, guild, "Member"):
```

Comment:
Future: consider writing a decorator to check if the user is a member to avoid repeating this auth check in a few different places.

Also, the role name should be a constant.

Reply:
I'd like to build out the `utils` folder with useful and properly setup decorators, but I was running out of steam to turn the _has_role_check() into a proper decorator. Will add an issue for it to do later.